File: middler.Common/Models/MiddlerRulePermission.cs
using middler.Common.Enums;

namespace middler.Common.Models
{
    public class MiddlerRulePermission
    {
        public string PrincipalName { get; set; }
        public PrincipalType Type { get; set; }
        public AccessMode AccessMode { get; set; }
        public string Client { get; set; }
        public string SourceAddress { get; set; }

    }

}

File: middlerApp.UI/projects/main/src/app/components/rules/models/middler-rule-permission.ts
export class MiddlerRulePermission {
    PrincipalName?: string;
    Type: PermissionType;
    AccessMode: AccessMode;
    Client?: string;
    SourceAddress?: string;
}

export type PermissionType = "User" | "Role" | "Authenticated" | "Everyone";
export type AccessMode = "Allow" | "Deny" | "Ignore";
